docs/@docs/label-rendering/component_diagram.mmd

```diff
@@ -0,0 +1,43 @@
+graph TB
+
+    subgraph harp-text-canvas
+    FontCatalog -.create.-> GlyphData
+    TextCanvas -.render.-> GlyphData
+    TextCanvas -.use.-> TextStyle["Text(Render|Layout)Style"]
+    end
+
+    subgraph harp-mapview
+    PoiManager -.create.-> PoiInfo
+    TileGeometryCreator -.create.-> Tile
+    TileGeometryCreator -.create.-> TextElement
+    TileGeometryCreator -.call.-> PoiManager
+    Tile --> TextElement
+    TextElement --> PoiInfo
+    TextElementsRenderer -.use.-> Tile
+    TextElementsRenderer -.call.-> PoiRenderer
+    PoiRenderer -.render.-> PoiInfo
+    end
+
+    subgraph harp-datasource-protocol
```

Comment:
I would like to see another subgraph which shows the TileDataSource, and the dependency it has with the Tile, i.e. the load method. You could also add the TileLoader and the dependency to the ITileDecoder. That would complete it a bit further.


, and the TileLoader calling the decodeTile on the ITileDecoder. You could also then

Reply:
I don't think that belongs to labeling documentation. I'd like to keep in that diagram strictly the classes and packages that have directly something to do with labels.
Classes derived from ITileDecoder decode labels, DecodedTile and Tile have members and methods related to labels, the Theme has label styling...
I do think it'd be good to have diagrams showing how tile decoding and loading works, but that belongs to another technical doc.

Reply:
Ok, I created a ticket to document the more general system: HARP-8700